app/services/find_or_register_user.rb

```diff
@@ -0,0 +1,27 @@
+class FindOrRegisterUser
+  attr_reader :github_id
+
+  def initialize(params)
+    @github_id = params[:github_id]
+  end
+
+  def perform
+    @_user = found_user = find_user
+
+    if found_user.nil?
+      @_user = register_user
+    end
+
+    @_user
+  end
+
+  private
+
+  def find_user
+    User.where(github_id: github_id).first
```

Comment:
there will always be a user because `from_oauth_hash(auth_hash)` creates a user from the oauth hash we get from github. https://github.com/18F/micropurchase/blob/e3e02f33c326142f2689e52282fb5c9f0e1f9a2b/app/models/user.rb#L14

move oauth logic into here?